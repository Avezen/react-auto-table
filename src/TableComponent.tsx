import React from 'react';
// @ts-ignore
import { TableNavigation } from './TableNavigation';
import {Table} from "./Table";

interface TableComponentProps {
  data?: any[];
  currentPage: number;
  setCurrentPage: any;
  modifyCurrentPage: any;
  pages: number;
  search: string;
  setSearch: any;
  sortBy: string;
  setSortBy: any;
  sortDirection: number;
  isFetching: boolean;
  error?: boolean;
  colWidths?: number[];
  tableHead: string[];
  currentlyOpenedTd: {
    id: null,
    key: null,
  };
  toggleTd: any;
}

export const TableComponent = ({
  data,
  currentPage,
  setCurrentPage,
  modifyCurrentPage,
  pages,
  search,
  setSearch,
  sortBy,
  setSortBy,
  sortDirection,
  isFetching,
  error,
  colWidths,
  tableHead,
  currentlyOpenedTd,
  toggleTd,
}: TableComponentProps) => (
  <React.Fragment>
    <div className={'table-nav--top'}>
      <TableNavigation setCurrentPage={setCurrentPage} modifyCurrentPage={modifyCurrentPage} currentPage={currentPage} pages={pages} />
      <input className={'input'} value={search} type={'text'} placeholder={'Search...'} onChange={setSearch} />
    </div>
    <Table
        data={data}
        sortBy={sortBy}
        setSortBy={setSortBy}
        sortDirection={sortDirection}
        isFetching={isFetching}
        tableHead={tableHead}
        colWidths={colWidths}
        currentlyOpenedTd={currentlyOpenedTd}
        toggleTd={toggleTd}
        error={error}
    />
    <div className={'table-nav--bottom'}>
      <TableNavigation setCurrentPage={setCurrentPage} modifyCurrentPage={modifyCurrentPage} currentPage={currentPage} pages={pages} />
    </div>
  </React.Fragment>
);
