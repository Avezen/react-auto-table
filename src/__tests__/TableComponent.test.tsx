import React from "react";
import {TableComponent, TableComponentProps} from "../TableComponent";
import renderer from 'react-test-renderer';

function renderTableComponent(props: Partial<TableComponentProps> = {}) {
    const defaultProps: TableComponentProps = {
        data: [
            {
                Id: 1,
                Name: 'Test 1'
            },
            {
                Id: 2,
                Name: 'Test 2'
            }
        ],
        currentPage: 1,
        setCurrentPage: () => {},
        modifyCurrentPage: () => {},
        pages: 2,
        search: '',
        setSearch: () => {},
        sortBy: 'Id',
        setSortBy: () => {},
        sortDirection: 1,
        isFetching: false,
        error: false,
        colWidths: [50, 25],
        tableHead: ['Id', 'Name'],
        currentlyOpenedTd: {
            id: null,
            key: null,
        },
        toggleTd: () => {},
        config: {}
    };

    return <TableComponent {...defaultProps} {...props} />;
}

describe("<TableComponent />", () => {
    test("should render component with data, with tableHead, with colWidths, sorted ASC by Id, with 2 pages on page 1", async () => {
        const findByTestId = renderTableComponent(),
            TableComponent = renderer.create(findByTestId).toJSON();

        expect(TableComponent).toMatchSnapshot();
    });
});
