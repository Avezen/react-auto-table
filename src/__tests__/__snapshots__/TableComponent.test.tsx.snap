// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableComponent /> should render component with data, with tableHead, with colWidths, sorted ASC by Id, with 2 pages on page 1 1`] = `
Array [
  <div
    className="table-nav--top"
  >
    <div
      className="table-navigation-pages"
    >
      <span
        data-testid="span"
      >
        Page 
        1
         of 
        2
        .
      </span>
      <div
        className="number-wrapper"
      >
        <input
          className="input number-input"
          data-testid="number-input"
          onChange={[Function]}
          type="number"
          value={1}
        />
        <div
          className="number-up"
          data-testid="number-up"
          onClick={[Function]}
        />
        <div
          className="number-down"
          data-testid="number-down"
          onClick={[Function]}
        />
      </div>
    </div>
    <input
      className="input"
      name="search"
      onChange={[Function]}
      placeholder="Search..."
      type="text"
      value=""
    />
  </div>,
  <table>
    <colgroup>
      <col
        width={50}
      />
      <col
        width={25}
      />
    </colgroup>
    <thead>
      <tr>
        <th
          onClick={[Function]}
        >
          Id
          <i
            className="arrow down"
          />
        </th>
        <th
          onClick={[Function]}
        >
          Name
        </th>
        <th>
          Actions
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <div
            className="truncated-string"
          >
            <div
              data-testid="truncated-value"
            >
              1
            </div>
          </div>
        </td>
        <td>
          <div
            className="truncated-string"
          >
            <div
              data-testid="truncated-value"
            >
              Test 1
            </div>
          </div>
        </td>
        <td>
          <button>
            Edit
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <div
            className="truncated-string"
          >
            <div
              data-testid="truncated-value"
            >
              2
            </div>
          </div>
        </td>
        <td>
          <div
            className="truncated-string"
          >
            <div
              data-testid="truncated-value"
            >
              Test 2
            </div>
          </div>
        </td>
        <td>
          <button>
            Edit
          </button>
        </td>
      </tr>
    </tbody>
  </table>,
  <div
    className="table-nav--bottom"
  >
    <div
      className="table-navigation-pages"
    >
      <span
        data-testid="span"
      >
        Page 
        1
         of 
        2
        .
      </span>
      <div
        className="number-wrapper"
      >
        <input
          className="input number-input"
          data-testid="number-input"
          onChange={[Function]}
          type="number"
          value={1}
        />
        <div
          className="number-up"
          data-testid="number-up"
          onClick={[Function]}
        />
        <div
          className="number-down"
          data-testid="number-down"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>,
]
`;
